integrationtest/integration_test.go

```diff
@@ -51,29 +83,97 @@ func TestUnauthorizedServiceAccountErrorCode(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	defer func() {
+	deferred := func() {
 		_, err = client.Logical().Delete("sys/auth/kubernetes")
 		if err != nil {
 			t.Fatal(err)
 		}
+	}
+
+	defer func() {
+		// just in case setupKubernetesAuth panics before returning deferred to the caller
+		if panicErr := recover(); panicErr != nil {
+			deferred()
+			panic(panicErr)
+		}
 	}()
 
-	_, err = client.Logical().Write("auth/kubernetes/config", map[string]interface{}{
-		"kubernetes_host": "https://kubernetes.default.svc.cluster.local",
-	})
+	if len(kubeConfigOverride) == 0 {
+		_, err = client.Logical().Write("auth/kubernetes/config", map[string]interface{}{
+			"kubernetes_host": "https://kubernetes.default.svc.cluster.local",
+		})
+	} else {
+		_, err = client.Logical().Write("auth/kubernetes/config", kubeConfigOverride)
+	}
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	_, err = client.Logical().Write("auth/kubernetes/role/test-role", map[string]interface{}{
-		"bound_service_account_names":      "test",
+		"bound_service_account_names":      boundServiceAccountName,
 		"bound_service_account_namespaces": "test",
 	})
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	_, err = client.Logical().Write("auth/kubernetes/login", map[string]interface{}{
+	return deferred, client
+}
+
+func TestSuccess(t *testing.T) {
+	deferred, client := setupKubernetesAuth(t, "vault", nil)
+	defer deferred()
+
+	_, err := client.Logical().Write("auth/kubernetes/login", map[string]interface{}{
+		"role": "test-role",
+		"jwt":  os.Getenv("KUBERNETES_JWT"),
+	})
+	if err != nil {
+		t.Fatalf("Expected successful login but got: %v", err)
+	}
+}
+
+func TestSuccessWithTokenReviewerJwt(t *testing.T) {
+	deferred, client := setupKubernetesAuth(t, "vault", map[string]interface{}{
+		"kubernetes_host":    "https://kubernetes.default.svc.cluster.local",
+		"token_reviewer_jwt": os.Getenv("TOKEN_REVIEWER_JWT"),
+	})
+	defer deferred()
+
+	_, err := client.Logical().Write("auth/kubernetes/login", map[string]interface{}{
+		"role": "test-role",
+		"jwt":  os.Getenv("KUBERNETES_JWT"),
+	})
+	if err != nil {
+		t.Fatalf("Expected successful login but got: %v", err)
+	}
+}
```

Comment:
It would be good to also test a login where `token_reviewer_jwt` is blank on the config, so the incoming `jwt` is also used to auth with the TokenReview API.